internal/backend/memory.go

```diff
@@ -8,117 +8,216 @@ import (
 	"errors"
 	"fmt"
 	"sync"
+	"time"
 )
 
 // StorageMemory is an implementation of the backend storage that stores data in memory
 type StorageMemory struct {
-	buckets map[string][]Object
+	buckets map[string]bucketInMemory
 	mtx     sync.RWMutex
 }
 
+type bucketInMemory struct {
+	Bucket
+	activeObjects   []Object
```

Comment:
rather than lists, we could use map to speed up access... but i dont think this project needs to care too much about efficiency, so keeping lists of objects as we used to have, for simplicity